pint/models/model_builder.py

```diff
@@ -90,7 +97,8 @@ def preprocess_parfile(self, parfile):
         self.param_inparF = param
         for key in repeat_par.keys():
             self.param_inparF[key + str(1)] = self.param_inparF.pop(key)
-        pfile.close()
+        if isinstance(parfile, str):
+            pfile.close()
```

Comment:
Closing files that were opened for reading is not essential, which is good because if an exception is raised this never gets called.